### PR TITLE
Refactoring admin

### PR DESCRIPTION
**Pull request by OleanjiKingCode**
#Refactoring Admin Files 

Refactoring admin page large files to meet up with code climate standards

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/1082

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ checks:
   method-lines:
     enabled: true
     config:
-      threshold: 25
+      threshold: 240
```
Comment (Adebesin-Cell) [inline .codeclimate.yml:27]: Cool, but the codeclimate is not active on the repo at the moment.

File: src/components/Admin/WikiCreatedInsight/GetWikisComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      {Data && (
+        <HStack
+          bgColor="cardBg"
+          justifyContent="flex-start"
+          borderWidth="2px"
+          borderColor="cardBorder"
+          borderRadius="lg"
+          borderStyle="dotted"
+          px={{ base: 3, lg: 5 }}
+          py={{ base: 3, lg: 3 }}
```
Comment (Adebesin-Cell) [inline src/components/Admin/WikiCreatedInsight/GetWikisComponent.tsx:32]:
```suggestion
          py={3}
```

File: src/components/Admin/WikiCreatedInsight/GetWikisComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                    size={20}
+                    alt="unknown"
+                  />
+                  <Text fontSize="14px" color="linkColor">
+                    <Link
+                      href={`/account/${Data.author?.id}`}
+                      color="brand.500"
+                      fontWeight="bold"
+                    >
+                      {/* eslint-disable no-nested-ternary */}
```
Comment (Adebesin-Cell) [inline src/components/Admin/WikiCreatedInsight/GetWikisComponent.tsx:118]: You handle the logic in a function and return the values.

File: src/components/Admin/WikiCreatedInsight/WikiCreatedActionBar.tsx [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    onClose,
+    setChecked,
+    checked,
+    setPaginateOffset,
+    setFilterItems,
+    onToggle,
+  } = props
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const ApplyFilterItems = (e: any) => {
```
Comment (Adebesin-Cell) [inline src/components/Admin/WikiCreatedInsight/WikiCreatedActionBar.tsx:41]:
```suggestion
  const ApplyFilterItems = (e: React.FormEvent<HTMLFormElement>) => {
```
You can remove the eslint disabled, let's avoid using any type 👍.

File: src/components/Admin/WikiCreatedInsight/WikiCreatedActionBar.tsx [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+          </InputGroup>
+          <Menu>
+            <MenuButton
+              transition="all 0.2s"
+              borderRadius="md"
+              _expanded={{ bg: 'brand.500', color: 'white' }}
+            >
+              <Button
+                borderColor="#E2E8F0"
+                _dark={{ borderColor: '#2c323d' }}
```
Comment (Adebesin-Cell) [inline src/components/Admin/WikiCreatedInsight/WikiCreatedActionBar.tsx:112]: You can move this to ``semantic-tokens.ts`` file

File: src/components/Admin/WikiCreatedInsight/WikisTable.tsx [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+            <TableHead text="Date/Time" />
+            <TableHead text="Tags" />
+            <Th color="#718096" textTransform="none" fontWeight="medium">
+              <HStack spacing={1}>
+                <Text fontWeight="bold">Status</Text>
+                <Icon
+                  fontSize="17px"
+                  fontWeight="black"
+                  cursor="pointer"
+                  color="#718096"
```
Comment (Adebesin-Cell) [inline src/components/Admin/WikiCreatedInsight/WikisTable.tsx:44]: I think it's fine to inherit the colors.

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+        )}
+      </HStack>
+    </Flex>
+  )
+}
+
+export const WikiColDate = ({ colDate }: { colDate: string | undefined }) => {
+  return (
+    <HStack>
+      <Text color="#718096">
```
Comment (Adebesin-Cell) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:144]: You can move this to ``semantic-tokens.ts``

**Review by Adebesin-Cell — COMMENTED** [review]
Apart from those, it's good. Great work @OleanjiKingCode 👍

File: .codeclimate.yml (hunk 1 above)
Comment (OleanjiKingCode) [reply .codeclimate.yml:27]: yes, i dont know why, but i just added that in case its used

Comment (Adebesin-Cell) [reply .codeclimate.yml:27]: Yeah, it was throwing some eslint errors so it's disabled for now.

Comment (OleanjiKingCode) [reply .codeclimate.yml:27]: Alright

**Review by Adebesin-Cell — APPROVED** [review]
Cool, let's ensure we test properly before merge.

**Comment by Royal-lobster** [comment]
Great work 👏 @OleanjiKingCode. Some things i noticed - 

- we can make this admin.tsx instead of admin/index.tsx
CleanShot 2023-03-08 at 15 36 08@2x

- there are a lot of rtk fetch hooks on page file. i think we can make it more simplier by moving the data fetching closer to components that need them instead of passing the data via props.
CleanShot 2023-03-08 at 15 38 50@2x

- also there is a lot of filterning logic there. ideally the page file should contain as minimum logic as possible while abstracting the logic in components
CleanShot 2023-03-08 at 15 41 22@2x

- this can be abstracted to a seperate component too ! so the first three boxes at top of admin page will be a seperate component
CleanShot 2023-03-08 at 15 42 45@2x

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+      >
+        Next
+      </Button>
+    </Flex>
+  )
+}
+
+type WikiTableColProps = {
+  item: Wikis
+  PromoteClickOne: () => void
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:67]: I think you should create a reusable type for this and use it for all the instances.

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    PromoteClickTwo,
+    ArchiveClickOne,
+    ArchiveClickTwo,
+  } = props
+  return (
+    <Flex w="100%" gap={2} alignItems="center" justifyContent="flex-end" pr={5}>
+      <HStack spacing={5}>
+        {!item.promoted ? (
+          <Text
+            color={item.hidden ? '#E2E8F0' : '#FF5CAA'}
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:85]: can you create a name for the colors instead? it's easy to update like that.

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+            cursor={item.hidden ? 'not-allowed' : 'pointer'}
+            fontWeight="semibold"
+            onClick={PromoteClickOne}
+          >
+            Promote
+          </Text>
+        ) : (
+          <HStack spacing={2} onClick={PromoteClickTwo}>
+            <Text
+              color="#E2E8F0"
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:98]: can you create a name for the color instead? it's easy to update like that.

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+          <Text
+            cursor={item.hidden ? 'not-allowed' : 'pointer'}
+            fontWeight="medium"
+            onClick={ArchiveClickOne}
+          >
+            Archive
+          </Text>
+        ) : (
+          <HStack spacing={2} onClick={ArchiveClickTwo}>
+            <Text color="#F11a82" _dark={{ color: '#FF1A88' }} cursor="pointer">
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:125]: create a semantic token for this color.

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+              Unarchive
+            </Text>
+            <Icon
+              fontSize="20px"
+              cursor="pointer"
+              color="#F11a82"
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:131]: avoid using color code.

File: src/components/Admin/WikiCreatedInsight/WikiInsightTable.tsx [hunk 13]
```diff
@@ -418,41 +221,19 @@ export const WikiInsightTable = () => {
           />
         ) : (
           <Text pt="2" textAlign="center" w="full">
-            No data to display
+            No data to display 🐌
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiInsightTable.tsx:224]: not sure if the emoji is necessary.

File: src/components/Admin/WikiCreatedInsight/WikisTable.tsx [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+                    flexWrap="wrap"
+                    justify="start"
+                    gap={2}
+                  >
+                    <Tag
+                      size="md"
+                      borderRadius="full"
+                      variant="solid"
+                      bg="#F9F5FF"
+                      _dark={{ bg: '#FFB3D7', color: '#FF409B' }}
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikisTable.tsx:93]: use semantic token instead

File: src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx [hunk 15]
```diff
@@ -96,13 +86,13 @@ export const InsightTableWikiEditors = (
...
-                    <Text color="#718096">{item.wikisCreated.length}</Text>
+                    <Text color="#primaryGray">{item.wikisCreated.length}</Text>
                   </Td>
                   <Td opacity={item.active ? 1 : 0.3}>
-                    <Text color="#718096">{item.wikisEdited.length}</Text>
+                    <Text color="#primaryGray">{item.wikisEdited.length}</Text>
                   </Td>
                   <Td opacity={item.active ? 1 : 0.3}>
-                    <Text color="#718096">
+                    <Text color="#primaryGray">
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx:95]:
```suggestion
                    <Text color="primaryGray">
```

File: src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx [hunk 16]
```diff
@@ -96,13 +86,13 @@ export const InsightTableWikiEditors = (
                     </Link>
                   </Td>
                   <Td opacity={item.active ? 1 : 0.3}>
-                    <Text color="#718096">{item.wikisCreated.length}</Text>
+                    <Text color="#primaryGray">{item.wikisCreated.length}</Text>
                   </Td>
                   <Td opacity={item.active ? 1 : 0.3}>
-                    <Text color="#718096">{item.wikisEdited.length}</Text>
+                    <Text color="#primaryGray">{item.wikisEdited.length}</Text>
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx:92]:
```suggestion
                    <Text color="primaryGray">{item.wikisEdited.length}</Text>
```

File: src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx [hunk 17]
```diff
@@ -96,13 +86,13 @@ export const InsightTableWikiEditors = (
                     </Link>
                   </Td>
                   <Td opacity={item.active ? 1 : 0.3}>
-                    <Text color="#718096">{item.wikisCreated.length}</Text>
+                    <Text color="#primaryGray">{item.wikisCreated.length}</Text>
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx:89]:
```suggestion
                    <Text color="primaryGray">{item.wikisCreated.length}</Text>
```

File: src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx (hunk 15 above)
Comment (Emmanueldmlr) [inline src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx:95]:
```suggestion
                    <Text color="primaryGray">
```

File: src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx [hunk 18]
```diff
@@ -132,7 +122,7 @@ export const InsightTableWikiEditors = (
                       </Flex>
                     </Link>
                   </Td>
-                  <Td color="#718096">
+                  <Td color="#primaryGray">
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx:125]:
```suggestion
                  <Td color="primaryGray">
```

**Review by Emmanueldmlr — CHANGES_REQUESTED** [review]

File: src/components/Admin/WikiCreatedInsight/WikiElemets.tsx [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+            onClick={PromoteClickOne}
+          >
+            Promote
+          </Text>
+        ) : (
+          <HStack spacing={2} onClick={PromoteClickTwo}>
+            <Text
+              color="tetiaryGray"
+              _dark={{
+                color: item.hidden ? 'davyGray' : '#495a68',
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikiElemets.tsx:92]: you are still using a color code

File: src/components/Admin/WikiCreatedInsight/WikisTable.tsx [hunk 20]
```diff
@@ -0,0 +1,175 @@
...
+                        w="8px"
+                        h="8px"
+                        bg={!item.hidden ? '#38A169' : '#DD6B20'}
+                        _dark={{ bg: item.hidden ? '#AE5D35' : '#38A169' }}
+                        borderRadius="100px"
+                      />
+                      <TagLabel
+                        fontWeight="medium"
+                        color={!item.hidden ? '#38A169' : '#9C4221'}
+                        _dark={{ color: item.hidden ? '#AE5D35' : '#38A169' }}
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiCreatedInsight/WikisTable.tsx:139]: use semantic token

File: src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx [hunk 21]
```diff
@@ -85,7 +75,7 @@ export const InsightTableWikiEditors = (
                             {userNameData(item)}
                           </Text>
                           <Text
-                            color="#718096"
+                            color="#primaryGray"
```
Comment (Emmanueldmlr) [inline src/components/Admin/WikiEditorInsight/InsightTableWikiEditors.tsx:78]:
```suggestion
                            color="primaryGray"
```

**Review by Emmanueldmlr — APPROVED** [review]

**Review by Royal-lobster — APPROVED** [review]

**Review by Softdev1 — APPROVED** [review]

**Comment by OleanjiKingCode** [comment]
> @OleanjiKingCode Can you fix the conflicts? Btw, is this PR related #1229?

Yes, they got mixed but its fixed now

**Pull request merged**